compiler/packages/babel-plugin-react-compiler/src/Inference/AnalyseFunctions.ts

```diff
@@ -108,7 +108,7 @@ function lower(func: HIRFunction): void {
   inferReferenceEffects(func, {isFunctionExpression: true});
   deadCodeElimination(func);
   inferMutableRanges(func);
-  leaveSSA(func);
+  rewriteInstructionKindsBasedOnReassignment(func);
```

Comment:
I noticed that we don't propagate phi types here -- do we not need them when analyzing functions?

Reply:
Good catch. The phi types really only affect later passes like MergeScopesThatInvalidateTogether, which we don't apply inside function expressions. That intuition is confirmed by there being basically no remaining output changes at this point that aren't accounted for, so i'm going to leave as-is and then we can follow-up to remove the PropagatePhiTypes pass and build that into InferTypes